From d216dc3ffe5c55295da04b023e4c32a1a3f3650e Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 7 Sep 2026 10:28:49 +0200
Subject: [PATCH 5/5] [lldb] avoid bare `#include Python.h`

otherwise, the following defines are not picked up before including `Python.h`
```
```
which affects whether non-stable-ABI symbols get used.
---
 .../Plugins/ScriptInterpreter/Python/PythonReadline.h      | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/lldb/source/Plugins/ScriptInterpreter/Python/PythonReadline.h b/lldb/source/Plugins/ScriptInterpreter/Python/PythonReadline.h
index c75219eb1a4f..98922568f703 100644
--- a/lldb/source/Plugins/ScriptInterpreter/Python/PythonReadline.h
+++ b/lldb/source/Plugins/ScriptInterpreter/Python/PythonReadline.h
@@ -9,14 +9,9 @@
 #ifndef LLDB_PLUGINS_SCRIPTINTERPRETER_PYTHON_PYTHONREADLINE_H
 #define LLDB_PLUGINS_SCRIPTINTERPRETER_PYTHON_PYTHONREADLINE_H
 
-#include "lldb/Host/Config.h"
+#include "lldb-python.h"
 
 #if LLDB_ENABLE_LIBEDIT && defined(__linux__)
-// NOTE: Since Python may define some pre-processor definitions which affect the
-// standard headers on some systems, you must include Python.h before any
-// standard headers are included.
-#include "Python.h"
-
 // no need to hack into Python's readline module if libedit isn't used.
 //
 #define LLDB_USE_LIBEDIT_READLINE_COMPAT_MODULE 1
